CONTRACTS: createDummyContract returns filename

Spartelfant opened the pull request:
# createDummyContract() method now returns filename…

…of generated dummy contract.

This tiny quality-of-life improvement means you don't have to search for the dummy contract you just created, instead its filename is simply returned by the method used to create it.

Spartelfant committed:
createDummyContract() now returns filename

createDummyContract() method now returns filename of generated dummy contract.

d0sboots commented:
Could you add the return to the function's docs (in ScriptEditor/something.d.ts) please?

Spartelfant commented:
I've updated the documentation!

I am sorry if I messed up the pull request, I thought I was adding the doc update to this one but it became its own pull request: #1130

Snarling commented:
First, thanks for contributing! But there are a few issues with this:

1. The changes should all be in the same PR, not spread across two different PRs.  You can do this by making another commit on your existing branch `patch-1`, instead of here where you made a new branch `patch-2` for the first attempt at documentation changes. Any commits to your `patch-1` branch will automatically be added to this PR.
2. You have a title error, that's just because we expect the PR title to be in a specific format. Something like `CATEGORY: Description of change` is expected, with an all-uppercase category with no spaces. At this time there are not strict rules on what categories are acceptable. For instance you could name your PR `CONTRACTS: createDummyContract returns a filename`.
3. On #1130 where you added the documentation change, you have a build error. This is due to the actual netscript function createDummyContract not returning a string even though your documentation says it is supposed to return a string. In addition to the generateDummyContract function you changed on this PR, you would need to change the NetscriptFunction createDummyContract in order for the new documentation to be correct.
4. Any time documentation is changed, it's also best practice to run `npm run doc` to regenerate new standalone documentation files. For this one you would definitely need a whole dev environment set up if you don't already.

If you'd like to fix this, I will let you try to figure it out. Let me know if you need any assistance with any of that.

Spartelfant commented:
Thanks @Snarling for your kind words and taking the time to explain how I can create a proper PR. I'll give it another go :)

Caldwell-74 commented:
just an fyi sometimes people forget to run `npm run doc ` or cant do  it 
so dont be surprised when you do it that a bunch of unrelated docs are changed aswell thats totaly fine

Spartelfant committed:
Update NetscriptDefinitions.d.ts

createDummyContract() returns a string (filename of contract)

Spartelfant committed:
Update CodingContract.ts

Return filename of created dummy contract

Spartelfant commented:
Alright I think that's the easy part done ;)
Now working on setting up a dev environment to be able to do the `npm run doc` part.

Spartelfant committed:
Update bitburner.codingcontract.createdummycontract.md

Updated documentation with `npm run doc` to reflect createDummyContract returning a string (filename of the created contract)

Spartelfant commented:
Hi @Snarling, I believe I have addressed all points you listed for me.

I'm looking forward to find out if I've succeeded in fabricating a proper pull request this time!

Snarling commented:
Looking at it on my phone, it looks good. I'll take a closer look in a little bit and merge it if there are no issues

Spartelfant commented:
Awesome, thanks again for your help ❤️
